src/include/codegen/parameter_cache.h

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// parameter_cache.h
+//
+// Identification: src/include/codegen/parameter_cache.h
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "codegen/codegen.h"
+#include "codegen/updateable_storage.h"
+#include "codegen/value.h"
+#include "expression/parameter.h"
+#include "type/type.h"
+
+namespace peloton {
+namespace codegen {
+
+//===----------------------------------------------------------------------===//
+// A parameter cache which stores the parameterized values for codegen runtime
+//===----------------------------------------------------------------------===//
+class ParameterCache {
+ public:
+  // Constructor
+  ParameterCache(const std::vector<expression::Parameter> &parameters) :
+      parameters_(parameters) {}
+
+  // Set the parameter values
+  void StoreValues(CodeGen &codegen, llvm::Value *query_parameters_ptr);
```

Comment:
Change this to `PopulateCache` or something. `StoreValues` doesn't really convey what's going on?